components/organisms/SearchDialog/search-dialog.tsx

```diff
@@ -23,6 +24,7 @@ const SearchDialog = () => {
   const setOpenSearch = store((state) => state.setOpenSearch);
   const debouncedSearchTerm = useDebounceTerm(searchTerm, 300);
   const [searchResult, setSearchResult] = useState<{ data: DbUserSearch[]; meta: {} }>();
+  const isMac = useOSDetection();
```

Comment:
In regards to your question about it being a hook or just a non-hook function, it would fail when rendering server-side with `ReferenceError: window is not defined`, so the custom hook works.